packages/nimble-components/src/select/index.ts

```diff
@@ -39,61 +89,776 @@ export class Select extends FoundationSelect implements ErrorPattern {
     @attr({ attribute: 'error-visible', mode: 'boolean' })
     public errorVisible = false;
 
+    @attr({ attribute: 'filter-mode' })
+    public filterMode: FilterMode = FilterMode.none;
+
+    /**
+     * Holds the current state for the calculated position of the listbox.
+     *
+     * @public
+     */
+    @observable
+    public position?: SelectPosition;
+
+    /**
+     * The ref to the internal `.control` element.
+     *
+     * @internal
+     */
+    @observable
+    public control!: HTMLElement;
+
+    /**
+     * Reference to the internal listbox element.
+     *
+     * @internal
+     */
+    public listbox!: HTMLDivElement;
+
+    /**
+     * The unique id for the internal listbox element.
+     *
+     * @internal
+     */
+    public listboxId: string = uniqueId('listbox-');
+
     /**
      * @internal
      */
     @observable
-    public region?: AnchoredRegion;
+    public scrollableElement!: HTMLElement;
+
+    /**
+     * @internal
+     */
+    @observable
+    public filterInputElement?: HTMLInputElement;
+
+    /**
+     * @internal
+     */
+    @observable
+    public region!: AnchoredRegion;
 
     /** @internal */
     @observable
     public hasOverflow = false;
 
-    // Workaround for https://github.com/microsoft/fast/issues/5123
-    public override setPositioning(): void {
+    /**
+     * @internal
+     */
+    @observable
+    public filteredOptions: ListboxOption[] = [];
+
+    /**
+     * @internal
+     */
+    @observable
+    public filter = '';
+
+    /**
+     * @internal
+     */
+    @observable
+    public scrollbarIsVisible = false;
+
+    /**
+     * @internal
+     */
+    @observable
+    public committedSelectedOption: ListboxOption | undefined = undefined;
+
+    /**
+     * The max height for the listbox when opened.
+     *
+     * @internal
+     */
+    @observable
+    public maxHeight = 0;
+
+    /**
+     * The component is collapsible when in single-selection mode with no size attribute.
+     *
+     * @internal
+     */
+    @volatile
+    public get collapsible(): boolean {
+        return !(this.multiple || typeof this.size === 'number');
+    }
+
+    private _value = '';
+    private forcedPosition = false;
+    private indexWhenOpened?: number;
+
+    // This intersection observer is to handle focus behavior for when the dropdown becomes
+    // visible.
+    private readonly regionElementIntersectionObserver: IntersectionObserver = new IntersectionObserver(
+        entries => {
+            if (
+                entries.length > 0
+                    && entries[entries.length - 1]!.intersectionRatio > 0
+            ) {
+                this.focusAndScrollOptionIntoView();
```

Comment:
No, if it is collapsed, this should be a no-op. If the popup is open this just means that if the selected option had been scrolled out of view, it will be scrolled into view, which seems like an okay behavior.